### DISCUSSION TITLE
Which version should I be using now (previously clblas)?

### DISCUSSION THREAD

**Discussion by stl3** [ggml-org/llama.cpp#8356] [Q&A] [answered]
I've been using llama-b3085-bin-win-clblast-x64, but in the latest versions I no longer see clblas, so I assume it has either been swapped with something better. My question is which one do I download now with similar functionality? I used clblas to split between my cpu (7950x3d) and gpu (4070S). Before when I was using clblas I was able to balance a nice usage of vram, sysram, gpu/cpu usage (wanted to avoid offloading everything to gpu, needed vram for other things). Would love some advice thank you.

**Comment by dspasyuk** [comment]
@stl3 The current version should work fine. Assuming you are using Linux:
`git clone https://github.com/ggerganov/llama.cpp.git; cd llama.cpp; sed -i 's/-arch=native/-arch=all/g' Makefile; make clean && LLAMA_CUDA=1 make -j 6;`
Use this manual for more details: https://github.com/ggerganov/llama.cpp/blob/master/examples/main/README.md

**Reply by stl3** [reply]
Ahh I do have linux machines but I'm actually running this on windows `llama-b3085-bin-win-clblast-x64`. Which current release should I use to replace `clblas`? 

```
cudart-llama-bin-win-cu11.7.1-x64.zip
cudart-llama-bin-win-cu12.2.0-x64.zip
llama-b3334-bin-macos-arm64.zip
llama-b3334-bin-macos-x64.zip
llama-b3334-bin-ubuntu-x64.zip
llama-b3334-bin-win-avx-x64.zip
llama-b3334-bin-win-avx2-x64.zip
llama-b3334-bin-win-avx512-x64.zip
llama-b3334-bin-win-cuda-cu11.7.1-x64.zip
llama-b3334-bin-win-cuda-cu12.2.0-x64.zip
llama-b3334-bin-win-kompute-x64.zip
llama-b3334-bin-win-llvm-arm64.zip
llama-b3334-bin-win-msvc-arm64.zip
llama-b3334-bin-win-noavx-x64.zip
llama-b3334-bin-win-openblas-x64.zip
llama-b3334-bin-win-rpc-x64.zip
llama-b3334-bin-win-sycl-x64.zip
llama-b3334-bin-win-vulkan-x64.zip

```
I want to split workload between gpu+vram and cpu+sysram like I did previously with `clblas` (am using 7950x3d cpu with 64GB sysram and 4070 Super with 12GB vram), but also able to limit `threads` and `ngl` like in clblas.

**Comment by Mar2ck** [comment]
Since you have a modern Nvidia gpu you should go with `win-cuda-cu12`, the CUDA backend can do everything clblast could and more. Make sure you install [CUDA toolkit](https://developer.nvidia.com/cuda-downloads) if you haven't already.

**Reply by stl3** [reply]
I don't mind trying cuda except it comes with an extra 3GB to download/install toolkit whereas running clblas I could just run it from the ~8MB download (~20MB decompressed) without installing anything extra.

**Reply by Mar2ck** [reply, marked as answer]
Vulkan should work without installing anything extra, it's much more lightweight but not as optimized as CUDA.
